<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<testsuite errors="0" failures="49" name="io.cucumber.core.plugin.JUnitFormatter" skipped="0" tests="58" time="4.625">
    <testcase classname="Authorization Failure Reporting" name="Publish simple message with valid ttl and auth permissions" time="1.957">
        <system-out>
            <![CDATA[Given I have a keyset with access manager enabled - without secret key......passed
Given a valid token with permissions to publish with channel 'channel-1'....passed
When I publish a message using that auth token with channel 'channel-1'.....passed
Then the result is successful...............................................passed
]]>
        </system-out>
    </testcase>
    <testcase classname="Authorization Failure Reporting" name="Publish simple message with expired auth token" time="0.091">
        <system-out>
            <![CDATA[Given I have a keyset with access manager enabled - without secret key......passed
Given an expired token with permissions to publish with channel 'channel-1'.passed
When I attempt to publish a message using that auth token with channel 'channel-1'.passed
Then an auth error is returned..............................................passed
* the error status code is 403..............................................passed
* the auth error message is 'Token is expired.'.............................passed
* the error service is 'Access Manager'.....................................passed
]]>
        </system-out>
    </testcase>
    <testcase classname="Authorization Failure Reporting" name="Publish simple message with invalid auth permissions" time="0.06">
        <system-out>
            <![CDATA[Given I have a keyset with access manager enabled - without secret key......passed
Given a valid token with permissions to publish with channel 'channel-1'....passed
When I attempt to publish a message using that auth token with channel 'channel-1'.passed
Then an auth error is returned..............................................passed
* the error status code is 403..............................................passed
* the auth error message is 'Forbidden'.....................................passed
* the error service is 'Access Manager'.....................................passed
]]>
        </system-out>
    </testcase>
    <testcase classname="Authorization Failure Reporting" name="Publish fails due to revoked token" time="0.057">
        <system-out>
            <![CDATA[Given I have a keyset with access manager enabled - without secret key......passed
Given a valid token with permissions to publish with channel 'channel-1'....passed
When I attempt to publish a message using that auth token with channel 'channel-1'.passed
Then an auth error is returned..............................................passed
* the error status code is 403..............................................passed
* the auth error message is 'Token is revoked.'.............................passed
* the error service is 'Access Manager'.....................................passed
]]>
        </system-out>
    </testcase>
    <testcase classname="Grant an access token" name="Attempt to grant an access token when ttl provided exceeds the max ttl configured (use default max 43200 for the test)" time="0.074">
        <system-out>
            <![CDATA[Given I have a keyset with access manager enabled...........................passed
Given the TTL 43201.........................................................passed
Given the 'channel-1' CHANNEL resource access permissions...................passed
* grant resource permission READ............................................passed
When I attempt to grant a token specifying those permissions................passed
Then an error is returned...................................................passed
* the error status code is 400..............................................passed
* the error message is 'Invalid ttl'........................................passed
* the error source is 'grant'...............................................passed
* the error detail message is 'Range should be 1 to 43200 minute(s).'.......passed
* the error detail location is 'ttl'........................................passed
* the error detail location type is 'body'..................................passed
]]>
        </system-out>
    </testcase>
    <testcase classname="Revoke an access token" name="Revoke existing valid token" time="0.036">
        <system-out>
            <![CDATA[Given I have a keyset with access manager enabled...........................passed
Given a token...............................................................passed
When I revoke a token.......................................................passed
Then I get confirmation that token has been revoked.........................passed
]]>
        </system-out>
    </testcase>
    <testcase classname="Revoke an access token" name="Revoke invalid token" time="0.058">
        <system-out>
            <![CDATA[Given I have a keyset with access manager enabled...........................passed
Given a token...............................................................passed
When I revoke a token.......................................................passed
Then an error is returned...................................................passed
* the error status code is 400..............................................passed
* the error message is 'Invalid token'......................................passed
* the error source is 'revoke'..............................................passed
* the error detail message is not empty.....................................passed
* the error detail location is 'token'......................................passed
* the error detail location type is 'path'..................................passed
* the error service is 'Access Manager'.....................................passed
]]>
        </system-out>
    </testcase>
    <testcase classname="Revoke an access token" name="Revoke a token while it is disabled on a server" time="0.048">
        <system-out>
            <![CDATA[Given I have a keyset with access manager enabled...........................passed
Given a token...............................................................passed
When I revoke a token.......................................................passed
Then an error is returned...................................................passed
* the error status code is 400..............................................passed
* the error message is 'Feature disabled'...................................passed
* the error source is 'revoke'..............................................passed
* the error detail message is 'Token revocation is disabled.'...............passed
* the error detail location is 'subscribe-key'..............................passed
* the error detail location type is 'path'..................................passed
* the error service is 'Access Manager'.....................................passed
]]>
        </system-out>
    </testcase>
    <testcase classname="Revoke an access token" name="Revoke a token with characters that require url encoding" time="0.04">
        <system-out>
            <![CDATA[Given I have a keyset with access manager enabled...........................passed
Given the token string 'unescaped-_.ABCabc123 escaped;,/?:@&=+$#'...........passed
When I revoke a token.......................................................passed
Then I get confirmation that token has been revoked.........................passed
]]>
        </system-out>
    </testcase>
    <testcase classname="Message encryption" name="Receiving an encrypted message with correct crypto key" time="0.042">
        <failure message="Scenario messageEncryption considered failure: &#10;                            pending - Subscribe request with timetoken zero,&#10;                            failed - " type="java.lang.AssertionError">
            <![CDATA[Given the crypto keyset.....................................................undefined
When I subscribe............................................................skipped
Then I receive the message in my subscribe response.........................skipped

StackTrace:
java.lang.AssertionError: Scenario messageEncryption considered failure: 
                            pending - Subscribe request with timetoken zero,
                            failed - 
	at org.junit.Assert.fail(Assert.java:89)
	at com.pubnub.contract.Hooks.after(Hooks.kt:43)
]]>
        </failure>
    </testcase>
    <testcase classname="Message encryption" name="Receiving an encrypted message with invalid crypto key" time="0.012">
        <failure message="Scenario messageEncryption considered failure: &#10;                            pending - Subscribe request with timetoken zero,&#10;                            failed - " type="java.lang.AssertionError">
            <![CDATA[Given the invalid-crypto keyset.............................................undefined
When I subscribe............................................................skipped
Then an error is thrown.....................................................skipped

StackTrace:
java.lang.AssertionError: Scenario messageEncryption considered failure: 
                            pending - Subscribe request with timetoken zero,
                            failed - 
	at org.junit.Assert.fail(Assert.java:89)
	at com.pubnub.contract.Hooks.after(Hooks.kt:43)
]]>
        </failure>
    </testcase>
    <testcase classname="Files" name="Listing files" time="0.034">
        <failure message="Scenario listingFiles considered failure: &#10;                            pending - List files request,&#10;                            failed - " type="java.lang.AssertionError">
            <![CDATA[Given the demo keyset.......................................................undefined
When I list files...........................................................skipped
Then I receive successful response..........................................skipped

StackTrace:
java.lang.AssertionError: Scenario listingFiles considered failure: 
                            pending - List files request,
                            failed - 
	at org.junit.Assert.fail(Assert.java:89)
	at com.pubnub.contract.Hooks.after(Hooks.kt:43)
]]>
        </failure>
    </testcase>
    <testcase classname="Files" name="Publishing file message" time="0.016">
        <failure message="Scenario publishingFileMessage considered failure: &#10;                            pending - Publish file message request,&#10;                            failed - " type="java.lang.AssertionError">
            <![CDATA[Given the demo keyset.......................................................undefined
When I publish file message.................................................skipped
Then I receive successful response..........................................skipped

StackTrace:
java.lang.AssertionError: Scenario publishingFileMessage considered failure: 
                            pending - Publish file message request,
                            failed - 
	at org.junit.Assert.fail(Assert.java:89)
	at com.pubnub.contract.Hooks.after(Hooks.kt:43)
]]>
        </failure>
    </testcase>
    <testcase classname="Files" name="Publishing file message failure" time="0.022">
        <failure message="Scenario publishingFileMessageFailure considered failure: &#10;                            pending - Publish file message request failure,&#10;                            failed - " type="java.lang.AssertionError">
            <![CDATA[Given the demo keyset.......................................................undefined
When I publish file message.................................................skipped
Then I receive error response...............................................skipped

StackTrace:
java.lang.AssertionError: Scenario publishingFileMessageFailure considered failure: 
                            pending - Publish file message request failure,
                            failed - 
	at org.junit.Assert.fail(Assert.java:89)
	at com.pubnub.contract.Hooks.after(Hooks.kt:43)
]]>
        </failure>
    </testcase>
    <testcase classname="Files" name="Deleting a file" time="0.015">
        <failure message="Scenario deletingFile considered failure: &#10;                            pending - Delete file request,&#10;                            failed - " type="java.lang.AssertionError">
            <![CDATA[Given the demo keyset.......................................................undefined
When I delete file..........................................................skipped
Then I receive successful response..........................................skipped

StackTrace:
java.lang.AssertionError: Scenario deletingFile considered failure: 
                            pending - Delete file request,
                            failed - 
	at org.junit.Assert.fail(Assert.java:89)
	at com.pubnub.contract.Hooks.after(Hooks.kt:43)
]]>
        </failure>
    </testcase>
    <testcase classname="Files" name="Downloading a file" time="0.013">
        <failure message="Scenario downloadingFile considered failure: &#10;                            pending - Download file request,&#10;                            failed - " type="java.lang.AssertionError">
            <![CDATA[Given the demo keyset.......................................................undefined
When I download file........................................................skipped
Then I receive successful response..........................................skipped

StackTrace:
java.lang.AssertionError: Scenario downloadingFile considered failure: 
                            pending - Download file request,
                            failed - 
	at org.junit.Assert.fail(Assert.java:89)
	at com.pubnub.contract.Hooks.after(Hooks.kt:43)
]]>
        </failure>
    </testcase>
    <testcase classname="History" name="Fetching message history" time="0.02">
        <failure message="Scenario fetchMessageHistory considered failure: &#10;                            pending - Fetch message history request,&#10;                            failed - " type="java.lang.AssertionError">
            <![CDATA[Given the demo keyset.......................................................undefined
When I fetch message history for single channel.............................skipped
Then I receive successful response..........................................skipped
And the response contains pagination info...................................skipped

StackTrace:
java.lang.AssertionError: Scenario fetchMessageHistory considered failure: 
                            pending - Fetch message history request,
                            failed - 
	at org.junit.Assert.fail(Assert.java:89)
	at com.pubnub.contract.Hooks.after(Hooks.kt:43)
]]>
        </failure>
    </testcase>
    <testcase classname="History" name="Fetching message history for multiple channels" time="0.016">
        <failure message="Scenario fetchMessageHistoryMulti considered failure: &#10;                            pending - Fetch message history with multiple channels request,&#10;                            failed - " type="java.lang.AssertionError">
            <![CDATA[Given the demo keyset.......................................................undefined
When I fetch message history for multiple channels..........................skipped
Then I receive successful response..........................................skipped

StackTrace:
java.lang.AssertionError: Scenario fetchMessageHistoryMulti considered failure: 
                            pending - Fetch message history with multiple channels request,
                            failed - 
	at org.junit.Assert.fail(Assert.java:89)
	at com.pubnub.contract.Hooks.after(Hooks.kt:43)
]]>
        </failure>
    </testcase>
    <testcase classname="History" name="Fetching message history with message actions" time="0.013">
        <failure message="Scenario fetchMessageHistoryActions considered failure: &#10;                            pending - Fetch message history actions with multiple channels request,&#10;                            failed - " type="java.lang.AssertionError">
            <![CDATA[Given the demo keyset.......................................................undefined
When I fetch message history with message actions...........................skipped
Then I receive successful response..........................................skipped

StackTrace:
java.lang.AssertionError: Scenario fetchMessageHistoryActions considered failure: 
                            pending - Fetch message history actions with multiple channels request,
                            failed - 
	at org.junit.Assert.fail(Assert.java:89)
	at com.pubnub.contract.Hooks.after(Hooks.kt:43)
]]>
        </failure>
    </testcase>
    <testcase classname="Message Actions" name="Adding a message action" time="0.019">
        <failure message="Scenario successfulAddMessageAction considered failure: &#10;                            pending - Add message action request,&#10;                            failed - " type="java.lang.AssertionError">
            <![CDATA[Given the demo keyset.......................................................undefined
When I add a message action.................................................skipped
Then I receive successful response..........................................skipped

StackTrace:
java.lang.AssertionError: Scenario successfulAddMessageAction considered failure: 
                            pending - Add message action request,
                            failed - 
	at org.junit.Assert.fail(Assert.java:89)
	at com.pubnub.contract.Hooks.after(Hooks.kt:43)
]]>
        </failure>
    </testcase>
    <testcase classname="Message Actions" name="Failing to add a message action" time="0.008">
        <failure message="Scenario failedAddMessageAction considered failure: &#10;                            pending - Add message action request,&#10;                            failed - " type="java.lang.AssertionError">
            <![CDATA[Given the demo keyset.......................................................undefined
When I add a message action.................................................skipped
Then I receive error response...............................................skipped

StackTrace:
java.lang.AssertionError: Scenario failedAddMessageAction considered failure: 
                            pending - Add message action request,
                            failed - 
	at org.junit.Assert.fail(Assert.java:89)
	at com.pubnub.contract.Hooks.after(Hooks.kt:43)
]]>
        </failure>
    </testcase>
    <testcase classname="Message Actions" name="Fetching message actions with pagination" time="0.007">
        <failure message="Scenario fetchMessageActions considered failure: &#10;                            pending - Fetch message action request,&#10;                            failed - " type="java.lang.AssertionError">
            <![CDATA[Given the demo keyset.......................................................undefined
When I fetch message actions................................................skipped
Then I receive successful response..........................................skipped
And the response contains pagination info...................................skipped

StackTrace:
java.lang.AssertionError: Scenario fetchMessageActions considered failure: 
                            pending - Fetch message action request,
                            failed - 
	at org.junit.Assert.fail(Assert.java:89)
	at com.pubnub.contract.Hooks.after(Hooks.kt:43)
]]>
        </failure>
    </testcase>
    <testcase classname="Message Actions" name="Deleting a message action" time="0.01">
        <failure message="Scenario successfulDeleteMessageAction considered failure: &#10;                            pending - Delete message action request,&#10;                            failed - " type="java.lang.AssertionError">
            <![CDATA[Given the demo keyset.......................................................undefined
When I delete a message action..............................................skipped
Then I receive successful response..........................................skipped

StackTrace:
java.lang.AssertionError: Scenario successfulDeleteMessageAction considered failure: 
                            pending - Delete message action request,
                            failed - 
	at org.junit.Assert.fail(Assert.java:89)
	at com.pubnub.contract.Hooks.after(Hooks.kt:43)
]]>
        </failure>
    </testcase>
    <testcase classname="Message Actions" name="Failing to delete a message action" time="0.021">
        <failure message="Scenario failedDeleteMessageAction considered failure: &#10;                            pending - Delete message action request,&#10;                            failed - " type="java.lang.AssertionError">
            <![CDATA[Given the demo keyset.......................................................undefined
When I delete a message action..............................................skipped
Then I receive error response...............................................skipped

StackTrace:
java.lang.AssertionError: Scenario failedDeleteMessageAction considered failure: 
                            pending - Delete message action request,
                            failed - 
	at org.junit.Assert.fail(Assert.java:89)
	at com.pubnub.contract.Hooks.after(Hooks.kt:43)
]]>
        </failure>
    </testcase>
    <testcase classname="Objects V2 UUID metadata" name="Get a UUID metadata for id" time="0.02">
        <failure message="src/test/resources/features/data/alice.json (No such file or directory)" type="java.io.FileNotFoundException">
            <![CDATA[Given I have a keyset with Objects V2 enabled...............................passed
Given the id for 'Alice' persona............................................failed
When I get the UUID metadata................................................skipped
Then I receive a successful response........................................skipped
And the UUID metadata for 'Alice' persona...................................skipped

StackTrace:
java.io.FileNotFoundException: src/test/resources/features/data/alice.json (No such file or directory)
	at java.base/java.io.FileInputStream.open0(Native Method)
	at java.base/java.io.FileInputStream.open(FileInputStream.java:219)
	at java.base/java.io.FileInputStream.<init>(FileInputStream.java:157)
	at kotlin.io.FilesKt__FileReadWriteKt.readText(FileReadWrite.kt:125)
	at com.pubnub.contract.UtilsKt.readPersonaFile(Utils.kt:15)
	at com.pubnub.contract.uuidmetadata.step.GivenSteps.the_uuid_for_persona(GivenSteps.kt:41)
	at ✽.the id for 'Alice' persona(file:///home/runner/work/kotlin/kotlin/sdk-specifications/features/objectsV2/uuid-metadata.feature:10)
]]>
        </failure>
    </testcase>
    <testcase classname="Objects V2 UUID metadata" name="Get a UUID with custom metadata, id stored in config" time="0.028">
        <failure message="src/test/resources/features/data/bob.json (No such file or directory)" type="java.io.FileNotFoundException">
            <![CDATA[Given I have a keyset with Objects V2 enabled...............................passed
Given current user is 'Bob' persona.........................................failed
When I get the UUID metadata with custom for current user...................skipped
Then I receive a successful response........................................skipped
And the UUID metadata for 'Bob' persona.....................................skipped

StackTrace:
java.io.FileNotFoundException: src/test/resources/features/data/bob.json (No such file or directory)
	at java.base/java.io.FileInputStream.open0(Native Method)
	at java.base/java.io.FileInputStream.open(FileInputStream.java:219)
	at java.base/java.io.FileInputStream.<init>(FileInputStream.java:157)
	at kotlin.io.FilesKt__FileReadWriteKt.readText(FileReadWrite.kt:125)
	at com.pubnub.contract.UtilsKt.readPersonaFile(Utils.kt:15)
	at com.pubnub.contract.uuidmetadata.step.GivenSteps.given_current_user_is_persona(GivenSteps.kt:42)
	at ✽.current user is 'Bob' persona(file:///home/runner/work/kotlin/kotlin/sdk-specifications/features/objectsV2/uuid-metadata.feature:17)
]]>
        </failure>
    </testcase>
    <testcase classname="Objects V2 UUID metadata" name="Set a UUID metadata" time="0.022">
        <failure message="src/test/resources/features/data/alice.json (No such file or directory)" type="java.io.FileNotFoundException">
            <![CDATA[Given I have a keyset with Objects V2 enabled...............................passed
Given the data for 'Alice' persona..........................................failed
When I set the UUID metadata................................................skipped
Then I receive a successful response........................................skipped
And the UUID metadata for 'Alice' persona contains updated..................skipped

StackTrace:
java.io.FileNotFoundException: src/test/resources/features/data/alice.json (No such file or directory)
	at java.base/java.io.FileInputStream.open0(Native Method)
	at java.base/java.io.FileInputStream.open(FileInputStream.java:219)
	at java.base/java.io.FileInputStream.<init>(FileInputStream.java:157)
	at kotlin.io.FilesKt__FileReadWriteKt.readText(FileReadWrite.kt:125)
	at com.pubnub.contract.UtilsKt.readPersonaFile(Utils.kt:15)
	at com.pubnub.contract.uuidmetadata.step.GivenSteps.the_data_for_persona(GivenSteps.kt:43)
	at ✽.the data for 'Alice' persona(file:///home/runner/work/kotlin/kotlin/sdk-specifications/features/objectsV2/uuid-metadata.feature:24)
]]>
        </failure>
    </testcase>
    <testcase classname="Objects V2 UUID metadata" name="Remove a UUID metadata for id" time="0.029">
        <failure message="src/test/resources/features/data/alice.json (No such file or directory)" type="java.io.FileNotFoundException">
            <![CDATA[Given I have a keyset with Objects V2 enabled...............................passed
Given the id for 'Alice' persona............................................failed
When I remove the UUID metadata.............................................skipped
Then I receive a successful response........................................skipped

StackTrace:
java.io.FileNotFoundException: src/test/resources/features/data/alice.json (No such file or directory)
	at java.base/java.io.FileInputStream.open0(Native Method)
	at java.base/java.io.FileInputStream.open(FileInputStream.java:219)
	at java.base/java.io.FileInputStream.<init>(FileInputStream.java:157)
	at kotlin.io.FilesKt__FileReadWriteKt.readText(FileReadWrite.kt:125)
	at com.pubnub.contract.UtilsKt.readPersonaFile(Utils.kt:15)
	at com.pubnub.contract.uuidmetadata.step.GivenSteps.the_uuid_for_persona(GivenSteps.kt:41)
	at ✽.the id for 'Alice' persona(file:///home/runner/work/kotlin/kotlin/sdk-specifications/features/objectsV2/uuid-metadata.feature:31)
]]>
        </failure>
    </testcase>
    <testcase classname="Objects V2 UUID metadata" name="Remove a UUID metadata, id stored in config" time="0.023">
        <failure message="src/test/resources/features/data/alice.json (No such file or directory)" type="java.io.FileNotFoundException">
            <![CDATA[Given I have a keyset with Objects V2 enabled...............................passed
Given current user is 'Alice' persona.......................................failed
When I remove the UUID metadata for current user............................skipped
Then I receive a successful response........................................skipped

StackTrace:
java.io.FileNotFoundException: src/test/resources/features/data/alice.json (No such file or directory)
	at java.base/java.io.FileInputStream.open0(Native Method)
	at java.base/java.io.FileInputStream.open(FileInputStream.java:219)
	at java.base/java.io.FileInputStream.<init>(FileInputStream.java:157)
	at kotlin.io.FilesKt__FileReadWriteKt.readText(FileReadWrite.kt:125)
	at com.pubnub.contract.UtilsKt.readPersonaFile(Utils.kt:15)
	at com.pubnub.contract.uuidmetadata.step.GivenSteps.given_current_user_is_persona(GivenSteps.kt:42)
	at ✽.current user is 'Alice' persona(file:///home/runner/work/kotlin/kotlin/sdk-specifications/features/objectsV2/uuid-metadata.feature:37)
]]>
        </failure>
    </testcase>
    <testcase classname="Objects V2 UUID metadata" name="Get all UUID metadata" time="0.046">
        <failure message="Unexpected request. Sequence Empty" type="com.pubnub.api.PubNubException">
            <![CDATA[Given I have a keyset with Objects V2 enabled...............................passed
When I get all UUID metadata................................................failed
Then I receive a successful response........................................skipped
And the UUID metadata for 'Alice' and 'James' persona.......................skipped

StackTrace:
PubNubException(errorMessage=Unexpected request. Sequence Empty, pubnubError=PubNubError(name=HTTP_ERROR, code=103, message='HTTP Error. Please check network connectivity. Please contact support with error details if the issue persists.'), jso=null, statusCode=500, affectedCall=retrofit2.OkHttpCall@884b93e)
	at com.pubnub.api.Endpoint.sync(Endpoint.kt:81)
	at com.pubnub.contract.uuidmetadata.step.WhenSteps.i_get_all_uuid_metadata(WhenSteps.kt:50)
	at ✽.I get all UUID metadata(file:///home/runner/work/kotlin/kotlin/sdk-specifications/features/objectsV2/uuid-metadata.feature:43)
]]>
        </failure>
    </testcase>
    <testcase classname="Objects V2 UUID metadata" name="Get all UUID metadata with custom" time="0.035">
        <failure message="Unexpected request. Sequence Empty" type="com.pubnub.api.PubNubException">
            <![CDATA[Given I have a keyset with Objects V2 enabled...............................passed
When I get all UUID metadata with custom....................................failed
Then I receive a successful response........................................skipped
And the UUID metadata for 'Bob' and 'Lisa' persona..........................skipped

StackTrace:
PubNubException(errorMessage=Unexpected request. Sequence Empty, pubnubError=PubNubError(name=HTTP_ERROR, code=103, message='HTTP Error. Please check network connectivity. Please contact support with error details if the issue persists.'), jso=null, statusCode=500, affectedCall=retrofit2.OkHttpCall@38f94a87)
	at com.pubnub.api.Endpoint.sync(Endpoint.kt:81)
	at com.pubnub.contract.uuidmetadata.step.WhenSteps.i_get_all_uuid_metadata_with_custom(WhenSteps.kt:55)
	at ✽.I get all UUID metadata with custom(file:///home/runner/work/kotlin/kotlin/sdk-specifications/features/objectsV2/uuid-metadata.feature:49)
]]>
        </failure>
    </testcase>
    <testcase classname="Publish a message" name="Publishing a message" time="0.016">
        <failure message="Scenario simplePublish considered failure: &#10;                            pending - Publish request with message,&#10;                            failed - " type="java.lang.AssertionError">
            <![CDATA[Given the demo keyset.......................................................undefined
When I publish a message....................................................skipped
Then I receive successful response..........................................skipped

StackTrace:
java.lang.AssertionError: Scenario simplePublish considered failure: 
                            pending - Publish request with message,
                            failed - 
	at org.junit.Assert.fail(Assert.java:89)
	at com.pubnub.contract.Hooks.after(Hooks.kt:43)
]]>
        </failure>
    </testcase>
    <testcase classname="Publish a message" name="Publishing a message with JSON metadata" time="0.015">
        <failure message="Scenario simplePublish considered failure: &#10;                            pending - Publish request with message,&#10;                            failed - " type="java.lang.AssertionError">
            <![CDATA[Given the demo keyset.......................................................undefined
When I publish a message with JSON metadata.................................skipped
Then I receive successful response..........................................skipped

StackTrace:
java.lang.AssertionError: Scenario simplePublish considered failure: 
                            pending - Publish request with message,
                            failed - 
	at org.junit.Assert.fail(Assert.java:89)
	at com.pubnub.contract.Hooks.after(Hooks.kt:43)
]]>
        </failure>
    </testcase>
    <testcase classname="Publish a message" name="Publishing a message with string metadata" time="0.021">
        <failure message="Scenario simplePublish considered failure: &#10;                            pending - Publish request with message,&#10;                            failed - " type="java.lang.AssertionError">
            <![CDATA[Given the demo keyset.......................................................undefined
When I publish a message with string metadata...............................skipped
Then I receive successful response..........................................skipped

StackTrace:
java.lang.AssertionError: Scenario simplePublish considered failure: 
                            pending - Publish request with message,
                            failed - 
	at org.junit.Assert.fail(Assert.java:89)
	at com.pubnub.contract.Hooks.after(Hooks.kt:43)
]]>
        </failure>
    </testcase>
    <testcase classname="Publish a message" name="Failing publish" time="0.021">
        <failure message="Scenario invalidPublish considered failure: &#10;                            pending - Publish request with message,&#10;                            failed - " type="java.lang.AssertionError">
            <![CDATA[Given the invalid keyset....................................................undefined
When I publish a message....................................................skipped
Then I receive error response...............................................skipped

StackTrace:
java.lang.AssertionError: Scenario invalidPublish considered failure: 
                            pending - Publish request with message,
                            failed - 
	at org.junit.Assert.fail(Assert.java:89)
	at com.pubnub.contract.Hooks.after(Hooks.kt:43)
]]>
        </failure>
    </testcase>
    <testcase classname="Signal" name="Sending a signal" time="0.017">
        <failure message="Scenario successfulSignal considered failure: &#10;                            pending - Signal request,&#10;                            failed - " type="java.lang.AssertionError">
            <![CDATA[Given the demo keyset.......................................................undefined
When I send a signal........................................................skipped
Then I receive successful response..........................................skipped

StackTrace:
java.lang.AssertionError: Scenario successfulSignal considered failure: 
                            pending - Signal request,
                            failed - 
	at org.junit.Assert.fail(Assert.java:89)
	at com.pubnub.contract.Hooks.after(Hooks.kt:43)
]]>
        </failure>
    </testcase>
    <testcase classname="Push Notifications" name="Listing push channels for GCM gateway" time="0.016">
        <failure message="Scenario listingPushChannels considered failure: &#10;                            pending - List push channels request,&#10;                            failed - " type="java.lang.AssertionError">
            <![CDATA[Given the demo keyset.......................................................undefined
When I list GCM push channels...............................................skipped
Then I receive successful response..........................................skipped

StackTrace:
java.lang.AssertionError: Scenario listingPushChannels considered failure: 
                            pending - List push channels request,
                            failed - 
	at org.junit.Assert.fail(Assert.java:89)
	at com.pubnub.contract.Hooks.after(Hooks.kt:43)
]]>
        </failure>
    </testcase>
    <testcase classname="Push Notifications" name="Listing push channels for APNS2 without topic" time="0.005">
        <failure message="The scenario has pending or undefined step(s)" type="java.lang.Exception">
            <![CDATA[Given the demo keyset.......................................................undefined
When I list APNS2 push channels.............................................skipped
Then I receive error response...............................................skipped
]]>
        </failure>
    </testcase>
    <testcase classname="Push Notifications" name="Listing push channels for APNS2" time="0.013">
        <failure message="Scenario listingPushChannels considered failure: &#10;                            pending - List push channels request,&#10;                            failed - " type="java.lang.AssertionError">
            <![CDATA[Given the demo keyset.......................................................undefined
When I list APNS2 push channels with topic..................................skipped
Then I receive successful response..........................................skipped

StackTrace:
java.lang.AssertionError: Scenario listingPushChannels considered failure: 
                            pending - List push channels request,
                            failed - 
	at org.junit.Assert.fail(Assert.java:89)
	at com.pubnub.contract.Hooks.after(Hooks.kt:43)
]]>
        </failure>
    </testcase>
    <testcase classname="Push Notifications" name="Adding push channels for GCM" time="0.007">
        <failure message="Scenario addingPushChannels considered failure: &#10;                            pending - Add push channels request,&#10;                            failed - " type="java.lang.AssertionError">
            <![CDATA[Given the demo keyset.......................................................undefined
When I add GCM push channels................................................skipped
Then I receive successful response..........................................skipped

StackTrace:
java.lang.AssertionError: Scenario addingPushChannels considered failure: 
                            pending - Add push channels request,
                            failed - 
	at org.junit.Assert.fail(Assert.java:89)
	at com.pubnub.contract.Hooks.after(Hooks.kt:43)
]]>
        </failure>
    </testcase>
    <testcase classname="Push Notifications" name="Adding push channels for FCM" time="0.012">
        <failure message="Scenario addingPushChannels considered failure: &#10;                            pending - Add push channels request,&#10;                            failed - " type="java.lang.AssertionError">
            <![CDATA[Given the demo keyset.......................................................undefined
When I add FCM push channels................................................skipped
Then I receive successful response..........................................skipped

StackTrace:
java.lang.AssertionError: Scenario addingPushChannels considered failure: 
                            pending - Add push channels request,
                            failed - 
	at org.junit.Assert.fail(Assert.java:89)
	at com.pubnub.contract.Hooks.after(Hooks.kt:43)
]]>
        </failure>
    </testcase>
    <testcase classname="Push Notifications" name="Adding push channels for APNS2 without topic" time="0.001">
        <failure message="The scenario has pending or undefined step(s)" type="java.lang.Exception">
            <![CDATA[Given the demo keyset.......................................................undefined
When I add APNS2 push channels..............................................skipped
Then I receive error response...............................................skipped
]]>
        </failure>
    </testcase>
    <testcase classname="Push Notifications" name="Adding push channels for APNS2" time="0.017">
        <failure message="Scenario addingPushChannels considered failure: &#10;                            pending - Add push channels request,&#10;                            failed - " type="java.lang.AssertionError">
            <![CDATA[Given the demo keyset.......................................................undefined
When I add APNS2 push channels with topic...................................skipped
Then I receive successful response..........................................skipped

StackTrace:
java.lang.AssertionError: Scenario addingPushChannels considered failure: 
                            pending - Add push channels request,
                            failed - 
	at org.junit.Assert.fail(Assert.java:89)
	at com.pubnub.contract.Hooks.after(Hooks.kt:43)
]]>
        </failure>
    </testcase>
    <testcase classname="Push Notifications" name="Removing push channels for GCM" time="0.016">
        <failure message="Scenario removingPushChannels considered failure: &#10;                            pending - Remove push channels request,&#10;                            failed - " type="java.lang.AssertionError">
            <![CDATA[Given the demo keyset.......................................................undefined
When I remove GCM push channels.............................................skipped
Then I receive successful response..........................................skipped

StackTrace:
java.lang.AssertionError: Scenario removingPushChannels considered failure: 
                            pending - Remove push channels request,
                            failed - 
	at org.junit.Assert.fail(Assert.java:89)
	at com.pubnub.contract.Hooks.after(Hooks.kt:43)
]]>
        </failure>
    </testcase>
    <testcase classname="Push Notifications" name="Removing push channels for APNS2 without topic" time="0.001">
        <failure message="The scenario has pending or undefined step(s)" type="java.lang.Exception">
            <![CDATA[Given the demo keyset.......................................................undefined
When I remove APNS2 push channels...........................................skipped
Then I receive error response...............................................skipped
]]>
        </failure>
    </testcase>
    <testcase classname="Push Notifications" name="Removing push channels for APNS2" time="0.018">
        <failure message="Scenario removingPushChannels considered failure: &#10;                            pending - Remove push channels request,&#10;                            failed - " type="java.lang.AssertionError">
            <![CDATA[Given the demo keyset.......................................................undefined
When I remove APNS2 push channels with topic................................skipped
Then I receive successful response..........................................skipped

StackTrace:
java.lang.AssertionError: Scenario removingPushChannels considered failure: 
                            pending - Remove push channels request,
                            failed - 
	at org.junit.Assert.fail(Assert.java:89)
	at com.pubnub.contract.Hooks.after(Hooks.kt:43)
]]>
        </failure>
    </testcase>
    <testcase classname="Push Notifications" name="Removing device for GCM" time="0.009">
        <failure message="Scenario removingDevice considered failure: &#10;                            pending - Remove device request,&#10;                            failed - " type="java.lang.AssertionError">
            <![CDATA[Given the demo keyset.......................................................undefined
When I remove GCM device....................................................skipped
Then I receive successful response..........................................skipped

StackTrace:
java.lang.AssertionError: Scenario removingDevice considered failure: 
                            pending - Remove device request,
                            failed - 
	at org.junit.Assert.fail(Assert.java:89)
	at com.pubnub.contract.Hooks.after(Hooks.kt:43)
]]>
        </failure>
    </testcase>
    <testcase classname="Push Notifications" name="Removing device for APNS2 without topic" time="0.009">
        <failure message="The scenario has pending or undefined step(s)" type="java.lang.Exception">
            <![CDATA[Given the demo keyset.......................................................undefined
When I remove APNS2 device..................................................skipped
Then I receive error response...............................................skipped
]]>
        </failure>
    </testcase>
    <testcase classname="Push Notifications" name="Removing device for APNS2" time="0.013">
        <failure message="Scenario removingDevice considered failure: &#10;                            pending - Remove device request,&#10;                            failed - " type="java.lang.AssertionError">
            <![CDATA[Given the demo keyset.......................................................undefined
When I remove APNS2 device with topic.......................................skipped
Then I receive successful response..........................................skipped

StackTrace:
java.lang.AssertionError: Scenario removingDevice considered failure: 
                            pending - Remove device request,
                            failed - 
	at org.junit.Assert.fail(Assert.java:89)
	at com.pubnub.contract.Hooks.after(Hooks.kt:43)
]]>
        </failure>
    </testcase>
    <testcase classname="Subscribe Loop auto-retry with access denied" name="don't auto-retry without auth key" time="0.011">
        <failure message="Scenario failedSubscribeWithInvalidAuthKey considered failure: &#10;                            pending - Subscribe request with timetoken zero,&#10;                            failed - " type="java.lang.AssertionError">
            <![CDATA[Given I have a keyset with access manager enabled...........................passed
Given no auth key...........................................................undefined
When I subscribe............................................................skipped
Then I receive access denied status.........................................skipped
* I don't auto-retry subscribe..............................................skipped

StackTrace:
java.lang.AssertionError: Scenario failedSubscribeWithInvalidAuthKey considered failure: 
                            pending - Subscribe request with timetoken zero,
                            failed - 
	at org.junit.Assert.fail(Assert.java:89)
	at com.pubnub.contract.Hooks.after(Hooks.kt:43)
]]>
        </failure>
    </testcase>
    <testcase classname="Subscribe Loop auto-retry with access denied" name="don't auto-retry with invalid PAMv2 auth key" time="0.019">
        <failure message="Scenario failedSubscribeWithInvalidAuthKey considered failure: &#10;                            pending - Subscribe request with timetoken zero,&#10;                            failed - " type="java.lang.AssertionError">
            <![CDATA[Given I have a keyset with access manager enabled...........................passed
Given auth key..............................................................undefined
When I subscribe............................................................skipped
Then I receive access denied status.........................................skipped
* I don't auto-retry subscribe..............................................skipped

StackTrace:
java.lang.AssertionError: Scenario failedSubscribeWithInvalidAuthKey considered failure: 
                            pending - Subscribe request with timetoken zero,
                            failed - 
	at org.junit.Assert.fail(Assert.java:89)
	at com.pubnub.contract.Hooks.after(Hooks.kt:43)
]]>
        </failure>
    </testcase>
    <testcase classname="Subscribe Loop auto-retry with access denied" name="don't auto-retry with invalid PAMv3 token" time="0.01">
        <failure message="Scenario failedSubscribeWithInvalidAuthKey considered failure: &#10;                            pending - Subscribe request with timetoken zero,&#10;                            failed - " type="java.lang.AssertionError">
            <![CDATA[Given I have a keyset with access manager enabled...........................passed
Given token.................................................................undefined
When I subscribe............................................................skipped
Then I receive access denied status.........................................skipped
* I don't auto-retry subscribe..............................................skipped

StackTrace:
java.lang.AssertionError: Scenario failedSubscribeWithInvalidAuthKey considered failure: 
                            pending - Subscribe request with timetoken zero,
                            failed - 
	at org.junit.Assert.fail(Assert.java:89)
	at com.pubnub.contract.Hooks.after(Hooks.kt:43)
]]>
        </failure>
    </testcase>
    <testcase classname="Subscribe Loop auto-retry with access denied" name="don't auto-retry with invalid secret key" time="0.014">
        <failure message="Scenario failedSubscribeWithInvalidAuthKey considered failure: &#10;                            pending - Subscribe request with timetoken zero,&#10;                            failed - " type="java.lang.AssertionError">
            <![CDATA[Given I have a keyset with access manager enabled...........................passed
Given secret key............................................................undefined
When I subscribe............................................................skipped
Then I receive access denied status.........................................skipped
* I don't auto-retry subscribe..............................................skipped

StackTrace:
java.lang.AssertionError: Scenario failedSubscribeWithInvalidAuthKey considered failure: 
                            pending - Subscribe request with timetoken zero,
                            failed - 
	at org.junit.Assert.fail(Assert.java:89)
	at com.pubnub.contract.Hooks.after(Hooks.kt:43)
]]>
        </failure>
    </testcase>
    <testcase classname="Subscribe Loop auto-retry with access denied" name="don't auto-retry with PAMv2 auth key permissions reset after subscribe" time="0.017">
        <failure message="Scenario failedSubscribeInvalidatedAuthKey considered failure: &#10;                            pending - Subscribe request with timetoken zero,&#10;                            failed - " type="java.lang.AssertionError">
            <![CDATA[Given I have a keyset with access manager enabled...........................passed
Given auth key..............................................................undefined
When I subscribe............................................................skipped
Then I receive the message in my subscribe response.........................skipped
* I receive access denied status............................................skipped
* I don't auto-retry subscribe..............................................skipped

StackTrace:
java.lang.AssertionError: Scenario failedSubscribeInvalidatedAuthKey considered failure: 
                            pending - Subscribe request with timetoken zero,
                            failed - 
	at org.junit.Assert.fail(Assert.java:89)
	at com.pubnub.contract.Hooks.after(Hooks.kt:43)
]]>
        </failure>
    </testcase>
    <testcase classname="Subscribe Loop auto-retry with access denied" name="don't auto-retry with PAMv3 token revoked after subscribe" time="0.015">
        <failure message="Scenario failedSubscribeInvalidatedAuthKey considered failure: &#10;                            pending - Subscribe request with timetoken zero,&#10;                            failed - " type="java.lang.AssertionError">
            <![CDATA[Given I have a keyset with access manager enabled...........................passed
Given token.................................................................undefined
When I subscribe............................................................skipped
Then I receive the message in my subscribe response.........................skipped
* I receive access denied status............................................skipped
* I don't auto-retry subscribe..............................................skipped

StackTrace:
java.lang.AssertionError: Scenario failedSubscribeInvalidatedAuthKey considered failure: 
                            pending - Subscribe request with timetoken zero,
                            failed - 
	at org.junit.Assert.fail(Assert.java:89)
	at com.pubnub.contract.Hooks.after(Hooks.kt:43)
]]>
        </failure>
    </testcase>
    <testcase classname="Subscribe Loop auto-retry with access denied" name="don't auto-retry with secret key invalidated after subscribe" time="0.011">
        <failure message="Scenario failedSubscribeInvalidatedAuthKey considered failure: &#10;                            pending - Subscribe request with timetoken zero,&#10;                            failed - " type="java.lang.AssertionError">
            <![CDATA[Given I have a keyset with access manager enabled...........................passed
Given secret key............................................................undefined
When I subscribe............................................................skipped
Then I receive the message in my subscribe response.........................skipped
* I receive access denied status............................................skipped
* I don't auto-retry subscribe..............................................skipped

StackTrace:
java.lang.AssertionError: Scenario failedSubscribeInvalidatedAuthKey considered failure: 
                            pending - Subscribe request with timetoken zero,
                            failed - 
	at org.junit.Assert.fail(Assert.java:89)
	at com.pubnub.contract.Hooks.after(Hooks.kt:43)
]]>
        </failure>
    </testcase>
    <testcase classname="Subscribe Loop" name="subscribe and recieve a published message" time="0.028">
        <failure message="Scenario simpleSubscribe considered failure: &#10;                            pending - Subscribe request with timetoken zero,&#10;                            failed - " type="java.lang.AssertionError">
            <![CDATA[Given the demo keyset.......................................................undefined
When I subscribe............................................................skipped
When I publish a message....................................................skipped
Then I receive the message in my subscribe response.........................skipped

StackTrace:
java.lang.AssertionError: Scenario simpleSubscribe considered failure: 
                            pending - Subscribe request with timetoken zero,
                            failed - 
	at org.junit.Assert.fail(Assert.java:89)
	at com.pubnub.contract.Hooks.after(Hooks.kt:43)
]]>
        </failure>
    </testcase>
    <testcase classname="Time" name="Getting PubNub time" time="0.008">
        <failure message="Scenario successfulTime considered failure: &#10;                            pending - Time request,&#10;                            failed - " type="java.lang.AssertionError">
            <![CDATA[When I request current time.................................................undefined
Then I receive successful response..........................................skipped

StackTrace:
java.lang.AssertionError: Scenario successfulTime considered failure: 
                            pending - Time request,
                            failed - 
	at org.junit.Assert.fail(Assert.java:89)
	at com.pubnub.contract.Hooks.after(Hooks.kt:43)
]]>
        </failure>
    </testcase>
</testsuite>
